dwave_networkx/algorithms/tsp.py

```diff
@@ -185,10 +195,10 @@ def traveling_salesperson_qubo(G, lagrange=None, weight='weight'):
             nextpos = (pos + 1) % N
 
             # going from u -> v
-            Q[((u, pos), (v, nextpos))] += G[u][v][weight]
+            Q[((u, pos), (v, nextpos))] += G[u][v].get(weight, missing_edge_weight)
```

Comment:
I think you're going to need something like `G[u].get(v, {}).get(weight, missing_edge_weight)` since either the weight attribute or the edge itself may be missing.